load_confounds/parser.py

```diff
@@ -76,34 +77,37 @@ def _load_high_pass(confounds_raw):
     high_pass_params = _find_confounds(confounds_raw, ["cosine"])
     return confounds_raw[high_pass_params]
 
-
-def _label_compcor(confounds_raw, compcor_suffix, n_compcor):
+def _label_compcor(confounds_json, compcor_suffix, n_compcor, acompcor_mask):
     """Builds list for the number of compcor components."""
-    compcor_cols = []
-    for nn in range(n_compcor + 1):
-        nn_str = str(nn).zfill(2)
-        compcor_col = compcor_suffix + "_comp_cor_" + nn_str
-        if compcor_col not in confounds_raw.columns:
-            warnings.warn(f"could not find any confound with the key {compcor_col}")
-        else:
-            compcor_cols.append(compcor_col)
-
+    compcor_cols = [key for key in confounds_json.keys() if compcor_suffix+'_comp' in key]
+    if compcor_suffix == 'a':
+        if acompcor_mask == 'combined':
+            compcor_cols = [key for key in compcor_cols if confounds_json[key]['Mask']=='combined']
+        else: compcor_cols = [key for key in compcor_cols if confounds_json[key]['Mask']!='combined']
+    if n_compcor != None:
+        if n_compcor > np.size(compcor_cols):
```

Comment:
This block could be turned into a small function  `_sanitize_n_compcor` called at the beginning. This will help readability.